src/person/serializers.py

```diff
@@ -45,32 +47,47 @@ class UserSerializer(serializers.ModelSerializer):
     interests = InterestSerializer(source="person.interests", many=True)
     groups = GroupSerializer(source="person.groups", many=True)
     has_onboarded = serializers.BooleanField(source="person.has_onboarded")
-    matches = serializers.SerializerMethodField("get_match_history")
+    current_match = serializers.SerializerMethodField("get_current_match")
+    previous_match = serializers.SerializerMethodField("get_previous_match")
 
     def get_goals(self, user):
         if user.person.goals is None:
-            return None
-        print(user.person.goals)
+            return []
         goals = json.loads(user.person.goals.replace("'", '"'))
-        return goals or None
+        return goals
+
+    def get_has_onboarded(self, user):
+        if user.person.has_onboarded is None:
+            return False
+        return user.person.has_onboarded
 
     def get_talking_points(self, user):
         if user.person.talking_points is None:
-            return None
+            return []
         talking_points = json.loads(user.person.talking_points.replace("'", '"'))
-        return talking_points or None
+        return talking_points
 
     def get_availability(self, user):
         if user.person.availability is None:
-            return None
+            return []
         availability = json.loads(user.person.availability.replace("'", '"'))
-        return availability or None
+        return availability
 
-    def get_match_history(self, user):
+    def get_current_match(self, user):
         matches = Match.objects.filter(Q(user_1=user) | Q(user_2=user)).order_by(
             "-created_date"
         )
-        return MatchSerializer(matches, user=user, many=True).data
+        if len(matches) == 0:
+            return None
+        return MatchSerializer(matches[0], user=user).data
+
+    def get_previous_match(self, user):
```

Comment:
ping me to talk about this, I know @lucyxubroad suggested we maintain the previous match but according to the Figma designs I'm seeing, it makes no reference to the previous match therefore we can probably get away without having to add this and having a route instead that gets the in-app feedback answers and updates the appropriate match

Reply:
@chalo2000  So the reason why I asked to include the previous match is that we check if a user should get the in app match feedback upon opening the app. So if a user does not have previous match, we just never pull up the feedback form but if they do have previous match and have not filled out the form, then we will pull up the feedback form to be filled out. Maybe we can get away with not including all the match information of previous match in the user response but it would be helpful for us to know if the user has submitted feedback on previous match and if they had a previous match?? so you return two boolean values to us?

@amycshuang I think amy would know more about this though!

Reply:
@lucyxubroad I see what you mean, I originally had something in mind where whenever the matches have been created for the next week we also flag users to receive the feedback form (which upon presenting on the frontend we can set the flag to false again after they have submitted feedback / closed the popup). That way we can keep the user model less bloated by having another boolean field instead of a match field with unused attributes.

For your concern about the two booleans, they'd only not have a previous match the first time they download the app since they haven't been matched with anyone so we can make a check on the amount of matches they have before flagging users to receive the feedback form.

I think this approach is better because the serialized user you get is smaller and all the logic you just mentioned we'd take care of on the backend for you so you just need to check a flag, similar to how we simplified matching by keeping it on the backend :) I think it would be great to dumb down the logic to a simple flag you can check in an if for whether or not to present something so you don't need to worry about all the edge cases since if we forget one instead of needing to handle it on the frontend side and go through a dreaded app review 😬 we can just quickly fix it on the backend and redeploy at moment's notice.

Let me know what you think / would prefer.